Source/Core/Core/PowerPC/PPCCache.cpp

```diff
@@ -168,11 +171,11 @@ void Cache::Invalidate(u32 addr)
   if (valid[set] & (1U << way))
   {
     if (addrs[set][way] & CACHE_VMEM_BIT)
-      lookup_table_vmem[(addrs[set][way] >> 5) & 0xfffff] = 0xff;
+      lookup_table_vmem[(addrs[set][way] >> 5) & (memory.GetFakeVMemMask() >> 5)] = 0xff;
```

Comment:
Perhaps just make local helpers like:

```cpp
static u32 GoodName(const Memory::MemoryManager& memory)
{
  return memory.GetFakeVMemMask() >> 5;
}
static u32 GoodName2(const Memory::MemoryManager& memory)
{
  return memory.GetExRamMask() >> 5;
}
static u32 GoodName3(const Memory::MemoryManager& memory)
{
  return memory.GetRamMask() >> 5;
}
```

so all the repeated shifting in 8 places isn't necessary

Reply:
IMO, the symmetry of the statement is pleasing.  If helpers were necessary, though, I believe a single helper which performs the right shift by 5 to align to an eight-word boundary would be all we need.  But then, any decent name describing this action will undoubtedly make the statement even wider than it already is.